Fix a few broken links in extensions documentation (#2226)
GoogleChrome/developer.chrome.com: site/en/docs/extensions/mv3/devtools/index.md

### Injecting a content script {: #injecting }

> The DevTools page can't call [`scripting.executeScript`][api-scripting-executescript] directly. To

Reviewer: ```suggestion
The DevTools page can't call [`scripting.executeScript()`][api-scripting-executescript] directly. To
```

Reviewer: Methods should have '()' after them. Please make sure this is consistent throughout.

inject a content script from the DevTools page, you must retrieve the ID of the inspected window's
tab using the [`inspectedWindow.tabId`][api-inspectedwindow-tabid] property and send a message to
the background page. From the background page, call
[`scripting.executeScript`][api-scripting-executescript] to inject the script.

{% Aside %}

If a content script has already been injected, you can add additional context scripts using the

> `eval` method. See [Passing the Selected Element to a Content Script][header-selected-element] for

Reviewer: ```suggestion
`eval()` method. See [Passing the Selected Element to a Content Script][header-selected-element] for
```

### Evaluating JavaScript in the inspected window {: #evaluating-js }

> You can use the [`inspectedWindow.eval`][api-inspectedwindow-eval] method to execute JavaScript code

Reviewer: ```suggestion
You can use the [`inspectedWindow.eval()`][api-inspectedwindow-eval] method to execute JavaScript code
```

> in the context of the inspected page. You can invoke the `eval` method from a DevTools page, panel

Reviewer: ```suggestion
in the context of the inspected page. You can invoke the `eval()` method from a DevTools page, panel
```

scripts.

The `eval` method is powerful when used in the right context and dangerous when used

> inappropriately. Use the [`scripting.executeScript`][api-scripting-executescript] method if you

Reviewer: ```suggestion
inappropriately. Use the [`scripting.executeScript()`][api-scripting-executescript] method if you
```

don't need access to the JavaScript context of the inspected page. For detailed cautions and a
comparison of the two methods, see [`inspectedWindow`][api-inspectedwindow].

### Passing the selected element to a content script {: #selected-element }

The content script doesn't have direct access to the current selected element. However, any code you

> execute using [`inspectedWindow.eval`][api-inspectedwindow-eval] has access to the DevTools console

Reviewer: ```suggestion
execute using [`inspectedWindow.eval()`][api-inspectedwindow-eval] has access to the DevTools console
```

and Console Utilities APIs. For example, in evaluated code you can use `$0` to access the selected
element.

To pass the selected element to a content script:

- Create a method in the content script that takes the selected element as an argument.

> - Call the method from the DevTools page using [`inspectedWindow.eval`][api-inspectedwindow-eval]

Reviewer: ```suggestion
- Call the method from the DevTools page using [`inspectedWindow.eval()`][api-inspectedwindow-eval]
```

### Getting a reference panel's `window` {: #panel-window }

> To `postMessage` from a devtools panel, you'll need a reference to its `window` object. Get a

Reviewer: ```suggestion
To call `postMessage()` from a devtools panel, you'll need a reference to its `window` object. Get a
```

Messaging between the DevTools page and content scripts is indirect, by way of the background page.

When sending a message _to_ a content script, the background page can use the

> [`tabs.sendMessage`][api-scripting-sendmessage] method, which directs a message to the content

Reviewer: ```suggestion
[`tabs.sendMessage()`][api-scripting-sendmessage] method, which directs a message to the content
```

### Messaging from injected scripts to the DevTools page {: #evaluated-scripts-to-devtools }

While the above solution works for content scripts, code that is injected directly into the page
(e.g. through appending a `<script>` tag or through

> [`inspectedWindow.eval`][api-inspectedwindow-eval]) requires a different strategy. In this context,

Reviewer: ```suggestion
[`inspectedWindow.eval()`][api-inspectedwindow-eval]) requires a different strategy. In this context,
```

> [`runtime.sendMessage`][api-runtime-sendmessage] will not pass messages to the background script as

Reviewer: ```suggestion
[`runtime.sendMessage()`][api-runtime-sendmessage] will not pass messages to the background script as
```

expected.

As a workaround, you can combine your injected script with a content script that acts as an
intermediary. To pass messages to the content script, you can use the

> [`window.postMessage`][mdn-postmessage] API. Here's an example, assuming the background script from

Reviewer: ```suggestion
[`window.postMessage()`][mdn-postmessage] method. Here's an example, assuming the background script from
```

Reviewer: API > method.

There's a best practice in this suggestion. Prefer the more specific term.

Your message will now flow from the injected script, to the content script, to the background
script, and finally to the DevTools page.

You can also consider [two alternative message passing techniques outlined
here][gh-devtools-messaging].

### Detecting when DevTools opens and closes {: #detecting-open-close }

If your extension needs to track whether the DevTools window is open, you can add an
[onConnect][api-runtime-onconnect] listener to the background page, and call

> [connect][api-runtime-connect] from the DevTools page. Since each tab can have its own DevTools

Reviewer: ```suggestion
[connect()][api-runtime-connect] from the DevTools page. Since each tab can have its own DevTools
```

Browse the source of these DevTools extension examples:

> - [Polymer Devtools Extension][gh-polymer-dt] - uses many helpers running in the host page to query

Reviewer: ```suggestion
- [Polymer Devtools Extension][gh-polymer-dt] - Uses many helpers running in the host page to query
```

  DOM/JS state to send back to the custom panel.

> - [React DevTools Extension][gh-react-dt] - Uses a submodule of Blink to reuse DevTools UI

Reviewer: ```suggestion
- [React DevTools Extension][gh-react-dt] - Uses a submodule of the renderer to reuse DevTools UI
```

Reviewer: This may not be the correct rewrite. Regardless, Chromium code names should be in external documents.
